src/main/java/com/es/phoneshop/model/product/DAO/ArrayListProductDao.java

```diff
@@ -0,0 +1,98 @@
+package com.es.phoneshop.model.product.DAO;
+
+import com.es.phoneshop.model.product.entity.Product;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Currency;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import java.util.stream.Collectors;
+
+public class ArrayListProductDao implements ProductDao {
```

Comment:
class name should not contain any details about the functional implementation
now you are using ArrayList as a storage, what if you decide to switch to another collection? Class name should describe the purpose, here we created a DAO for handling products
Name should be smth like "CustomProductDao" or "ProductDaoImpl"